service/src/test/java/bio/terra/workspace/app/controller/WorkspaceApiControllerConnectedTest.java

```diff
@@ -247,6 +256,59 @@ public void listWorkspaces_requesterIsDiscoverer_requestMinHighestRoleNotSet_ret
     assertTrue(listedWorkspaces.isEmpty());
   }
 
+  @Test
+  public void grantRole_logsAnActivity() throws Exception {
+    mockMvcUtils.grantRole(
+        userAccessUtils.defaultUserAuthRequest(),
+        workspace.getId(),
+        WsmIamRole.DISCOVERER,
+        userAccessUtils.getSecondUserEmail());
+
+    ActivityLogChangeDetails changeDetails = getLastChangeDetails(workspace.getId());
+    assertEquals(userAccessUtils.getSecondUserEmail(), changeDetails.changeSubjectId());
+    assertEquals(WsmObjectType.USER, changeDetails.changeSubjectType());
+    assertEquals(userAccessUtils.getDefaultUserEmail(), changeDetails.actorEmail());
+    assertEquals(OperationType.GRANT_WORKSPACE_ROLE, changeDetails.operationType());
+  }
+
+  @Test
+  public void removeRole_logsAnActivity() throws Exception {
+    mockMvcUtils.grantRole(
+        userAccessUtils.defaultUserAuthRequest(),
+        workspace.getId(),
+        WsmIamRole.DISCOVERER,
+        userAccessUtils.getSecondUserEmail());
+
+    mockMvcUtils.removeRole(
+        userAccessUtils.defaultUserAuthRequest(),
+        workspace.getId(),
+        WsmIamRole.DISCOVERER,
+        userAccessUtils.getSecondUserEmail());
+
+    ActivityLogChangeDetails changeDetails = getLastChangeDetails(workspace.getId());
+    assertEquals(userAccessUtils.getSecondUserEmail(), changeDetails.changeSubjectId());
+    assertEquals(WsmObjectType.USER, changeDetails.changeSubjectType());
+    assertEquals(userAccessUtils.getDefaultUserEmail(), changeDetails.actorEmail());
+    assertEquals(OperationType.REMOVE_WORKSPACE_ROLE, changeDetails.operationType());
+  }
+
+  /**
```

Comment:
Are we exposing this information in the API? If so, I'd rather test the API. Our controller tests usually call the API.

If not, add comment

```
// We don't expose this in API yet, so read from DB.
// TODO(PF-XXX): Change to call API.
```